lib/ipmi_backend_utils.pm

```diff
@@ -157,6 +158,9 @@ sub setup_console_in_grub {
     }
     save_screenshot;
     upload_logs($grub_cfg_file);
+    if (is_remote_backend && check_var('ARCH', 'aarch64') && (get_var("VIRT_PRJ2_HOST_UPGRADE") || get_var("VIRT_PRJ4_GUEST_UPGRADE"))) {
```

Comment:
this does not look like good design. You can do that freely in test modules but as this is a "lib" module I recommend to keep it out. Because you do it at the end of `setup_console_in_grub` it should not be too hard to cover this from the test module side to call wherever necessary, of course without duplicating it multiple times. If necessary you could wrap the method `setup_console_in_grub` in the base class of the necessary test modules, e.g. "virt_autotest_base"?

Reply:
This only needs to be called from update_package module. I put it back into update_package and refined IF decision conditions. I think it can be merged.

Reply:
I agree with okurz mostly. Good design is important. Able to be reused and no duplication is important.   Also we need to always keep future needs in mind.  What if we have future more needs on this, which is highly possible? So wrapping this in a function is suggested.

IMHO this need to use vendor-specific serial console should be not only for offline upgrade. It actually is needed for all your kvm on arm tests. Serial console should be always correctly configured. Correct me if any reason against it. Essentially, what this code does is to set serial console correctly. This follows the purpose of function setup_console_in_grub. If you can make it not break existing functions and tests, making it here perfectly is a good way.